### ISSUE TITLE
Tutorial: Classifying Names in model's forward

### ISSUE THREAD

**Issue by zplovekq** [pytorch/fairseq#1839]
In the model's ,it says "WARNING: The inputs have padding, so we should mask those  elements here so that padding doesn't affect the results."
So i modify the code:
   

> ```
>         real_output=torch.zeros(bsz,len(self.output_vocab)).to(hidden.device)
>         for i in range(len(src_lengths)):
>             if i == len(src_lengths)-1:
>                 continue
>             assert src_lengths[i]>=src_lengths[i+1]
>         for i in range(max_src_len):
>             real_batch=sum(src_lengths>i)
>             input = self.one_hot_inputs[src_tokens[:real_batch, i].long()]
>             hidden=hidden[:real_batch]
>             output, hidden = self.rnn(input, hidden)
>             real_output[:real_batch]=output
>         return real_output  
> ```
>
> 
But i see the training not stop. Like this:
![image](https://user-images.githubusercontent.com/39880124/76678342-ee7f0380-6611-11ea-8943-f73b411957da.png)
Thanks for help!

**Comment by myleott** [comment]
Your code snippet looks good.

> But i see the training not stop. Like this:

By default fairseq will keep training indefinitely. You can add `--max-epoch=10` to train for 10 epochs. Alternatively you could set an early stopping patience with `--patience 2`, which will terminate after the validate loss doesn't go up for 2 consecutive epochs.
